File: CRM.Api/Models/Group.cs
using System.ComponentModel.DataAnnotations;

namespace CRM.Api.Models
{
    public class Group
    {
        public int Id { get; set; }

        [Required]
        [StringLength(100)]
        public string Name { get; set; } = string.Empty;

        public string? Description { get; set; }

        // Hierarchy support
        public int? ParentGroupId { get; set; }
        public Group? ParentGroup { get; set; }
        public ICollection<Group> SubGroups { get; set; } = new List<Group>();

        // Membership
        public bool IsDynamic { get; set; } = false;
        public string? DynamicQuery { get; set; } // Stores filter logic
        
        // Additional fields for UI
        public string? Category { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }

        // Relationship: Many-to-Many with Contacts
        public ICollection<Contact> Contacts { get; set; } = new List<Contact>();
    }
}

File: CRM.Web/src/types/group.ts
import type { Contact } from './contact';

export interface Group {
    id: number;
    name: string;
    description?: string;

    // Hierarchy
    parentGroupId?: number;
    parentGroup?: Group;
    subGroups?: Group[];

    // Dynamic group support
    isDynamic: boolean;
    dynamicQuery?: string;

    // Category for organization
    category?: string;

    // Timestamps
    createdAt?: string;
    updatedAt?: string;

    // Members
    contacts?: Contact[];
}

// Dynamic query definition for smart groups
export interface DynamicQueryDefinition {
    filters: QueryFilter[];
}

export interface QueryFilter {
    field: string;
    operator: 'equals' | 'contains' | 'startsWith' | 'endsWith' | 'greaterThan' | 'lessThan';
    value: string;
}

// Available fields for dynamic filtering
export const FILTERABLE_FIELDS = [
    { value: 'city', label: 'City', type: 'text' },
    { value: 'state', label: 'State', type: 'text' },
    { value: 'company', label: 'Company', type: 'text' },
    { value: 'jobTitle', label: 'Job Title', type: 'text' },
    { value: 'leadSource', label: 'Lead Source', type: 'select' },
    { value: 'status', label: 'Status', type: 'select' },
];

// Group categories
export const GROUP_CATEGORIES = [
    'Marketing',
    'Sales',
    'Support',
    'Events',
    'Partners',
    'Vendors',
    'Internal',
    'Other'
];
